### ISSUE TITLE
Modbus TCP/IP server keeps spinning in a loop after the client disconnects

### ISSUE THREAD

**Issue by hmarius** [AndreasAmMueller/Modbus#9]
After the client disconnects the HandleClient method in ModbusTcpServer keeps looping with high cpu usage.
Please see the following test case: https://github.com/hmarius/ModbusTestCase

The problem may be at those lines:
```
var count = await stream.ReadAsync(buffer, 0, buffer.Length);
requestBytes.AddRange(buffer.Take(count));
```
Note: [link](https://stackoverflow.com/questions/34685670/tcpclient-networkstream-readasync-c-sharp)

Thank you!

**Comment by AndreasAmMueller** [comment]
I see, thank you for that Link. Will fix that :-)

**Comment by hmarius** [comment]
That will be great!
While you are at it :) please take in consideration the case when the received packet is incomplete like in this situation:
![image](https://user-images.githubusercontent.com/8322949/74667653-5590d000-51ac-11ea-8414-bab689712c96.png)
The incomplete packet is: 00 5C 00 00 00 06 01 03 00 00 00 [0a missing]

**Comment by AndreasAmMueller** [comment]
Should be fixed with v1.0.0, otherwise reopen.
